router/router.go

```diff
@@ -1544,21 +1529,49 @@ func (rt *HandleT) commitStatusList(responseList *[]jobResponseT) {
 			status := resp.status.JobState
 			userID := resp.userID
 			worker := resp.worker
+			rt.eventOrderSyncMu.Lock()
+
 			if status == jobsdb.Succeeded.State || status == jobsdb.Aborted.State {
 				worker.failedJobIDMutex.RLock()
 				lastJobID, ok := worker.failedJobIDMap[userID]
 				worker.failedJobIDMutex.RUnlock()
 				if ok && lastJobID == resp.status.JobID {
-					rt.toClearFailJobIDMutex.Lock()
 					rt.logger.Debugf("[%v Router] :: clearing failedJobIDMap for userID: %v", rt.destName, userID)
 					_, ok := rt.toClearFailJobIDMap[worker.workerID]
 					if !ok {
 						rt.toClearFailJobIDMap[worker.workerID] = make([]string, 0)
 					}
 					rt.toClearFailJobIDMap[worker.workerID] = append(rt.toClearFailJobIDMap[worker.workerID], userID)
-					rt.toClearFailJobIDMutex.Unlock()
 				}
 			}
+
+			// aborted jobIDs
+			worker.abortedUserMutex.RLock()
+			if runningJobs, ok := worker.abortedUserIDMap[userID]; ok {
+				if status == jobsdb.Succeeded.State {
+					// clear the map in the next generatorLoop
+					_, ok := rt.toClearAbortedJobIDMap[worker.workerID]
+					if !ok {
+						rt.toClearAbortedJobIDMap[worker.workerID] = make([]string, 0)
+					}
+					rt.toClearAbortedJobIDMap[worker.workerID] = append(rt.toClearAbortedJobIDMap[worker.workerID], userID)
+				} else { // any other state
+					if _, ok = runningJobs[resp.status.JobID]; ok {
+						// remove the jobID from the concurrency map during the next generatorLoop
```

Comment:
Why would these job ID's even end up in the concurrency map?

Reply:
Yeah I got it

Reply:
if they entered the pipeline while the concurrency limiter was on, they were added in router.go#1306